docs/scripting-element-object.rst

```diff
@@ -0,0 +1,566 @@
+``shot`` is a JPEG screenshot data, as a :ref:`binary object <binary-objects>`.
+When the result is empty (e.g. if the element is not visible) ``nil`` is returned.
+
+**Async:** no.
+
+*pad* parameter sets the padding of the resulting image. If it is a single integer then the
+padding from all sides will be equal. If the value of the padding is positive the resulting screenshot
+will be expanded by the specified amount of pixes. And if the value of padding is negative the resulting
+screenshot will be shrunk by the specified amount of pixes.
+
+See more in :ref:`splash-jpeg`.
+
+
+.. _splash-element-visible:
+
+element:visible
+---------------
+
+Check whether the element is visible.
+
+**Signature:** ``visible = element:visible()``
+
+**Returns:** ``visible`` indicated whether
+the element is visible.
+
+**Async:** no.
+
+
+.. _splash-element-text:
+
+element:text
+------------
+
+Fetch a text information from the element
+
+**Signature:** ``text = element:text()``
+
+**Returns:** ``text`` is a text content
+of the element.
+
+**Async:** no.
+
+It tries to return the trimmed value of the following JavaScript ``Node`` properties:
+
+* textContent
+* innerText
+* value
+
+If all of them are empty an empty string is returned.
+
+
+.. _splash-element-info:
+
+element:info
+------------
+
+Get useful information about the element.
+
+**Signature:** ``info = element:info()``
+
+**Returns:** ``info`` is a table with element info.
+
+**Async:** no.
+
+Info is a table with the following fields:
+
+* nodeName - node name in a lower case (e.g. *h1*)
+* attributes - table with attributes names and its values
+* tag - html string representation of the element
+* html - inner html of the element
+* text - inner text of the element
+* x - x coordinate of the element
+* y - y coordinate of the element
+* width - width of the element
+* height - height of the element
+* visible - flag representing if the element is visible
+
+
+.. _splash-element-field-value:
+
+element:field_value
+-------------------
+
+Get value of the field element (input, select).
+
+**Signature:** ``ok, info = element:field_value()``
+
+**Returns:** ``ok, value`` pair. If ``ok`` is nil then error happened during the function call;
+``value`` provides an information about error type; otherwise ``value`` is a value of the
+element.
+
+**Async:** no.
+
+The value can be a
+* string - for text/radio input and select and other element types
+* array of strings - for multi select
+* boolean - for checkbox input
+
+
+.. _splash-element-form-values:
+
+element:form_values
+-------------------
+
+Return a table with form values if the element type is *form*
+
+**Signature:** ``ok, info = element:form_values()``
+
+**Returns:** ``ok, values`` pair. If ``ok`` is nil then error happened during the function call
+or node type is not *form*; ``values`` provides an information about error type; otherwise
+``values`` is a table of values.
```

Comment:
Is it just a table of values (i.e. a plain array with values), or a table with element names as keys and values as values?